internal/extsvc/github/globallock.go

```diff
@@ -63,7 +65,14 @@ func restrictGitHubDotComConcurrency(logger log.Logger, doer httpcli.Doer, r *ht
 		metricFailedLockRequestsGauge.Inc()
 		// Note that we do NOT fail the request here, this lock is considered best
 		// effort.
-		logger.Error("failed to get mutex for GitHub.com, concurrent requests may occur and rate limits can happen", log.Error(err))
+		//
+		// We log a warning if the error is ErrTaken, since this can happen from time to time.
```

Comment:
some addtl info so we don't forget
```suggestion
		// We log a warning if the error is ErrTaken, since this can happen from time to time. It means that we didn't get the global lock in the permitted number of tries. Instead of blocking indefinitely, we let the request pass.
```